OPCB/functions/missions/artillery.sqf

```diff
@@ -1,36 +1,36 @@
-params ["_village", "_centerPos"];
-_stations = [];
-_artgroups = [];
-_base = [0,0,0];
-_group = createGroup [resistance,true];
-[_group] call CHAB_fnc_serverGroups;
-_comps = ["mortar1","mortar2","mortar3"];
+params ["_village", "_centerUnit"];
+private _artgroups = [];
+private _base = [0,0,0];
+private _comps = ["mortar1","mortar2","mortar3"];
+private _mortars = ["rhs_2b14_82mm_vdv"];
+private _stations = [];
 	
 for "_i" from 0 to 3 do {
+	_base = [20, getpos _centerUnit, 2000] call CHAB_fnc_findSpot;
+	private _dir = [ _base, _village ] call BIS_fnc_dirTo;
 
-	_markpos = globalWaterPos;
-	while { surfaceIsWater _markpos || (_markpos distance officer_jeff <1000) || surfaceIsWater _base || _base distance2D _centerPos < 400} do {
-		_markpos = _village getPos[random [500,1000,1500],random 360];
-		_base = [_markpos,15,500,10, 0, 0.5, 0,[],[globalWaterPos,globalWaterPos]] call BIS_fnc_findSafePos;
-	};
-	
-	_civilian = _group createUnit ["C_IDAP_Man_AidWorker_01_F", _base, [], 2, "NONE"];
-	_dir = [ _civilian, _village ] call BIS_fnc_dirTo;
+	private _comp = [selectRandom _comps, _base, [0,0,0], _dir, true, true ] call LARs_fnc_spawnComp;
+
+	private _artiPos = [_comp] call CHAB_fnc_findFlowerPots;
 
-	_pos = getPos _civilian;
-	_comp = [selectRandom _comps,_pos, [0,0,0], _dir, true, true ] call LARs_fnc_spawnComp;
+	private _spawnedGroup = createGroup [resistance, true];
+
+	[_spawnedGroup] call CHAB_fnc_serverGroups;
+	([_artiPos, random 360, selectRandom _mortars, _spawnedGroup] call BIS_fnc_spawnVehicle) params ["_createdVehicle", "_crew"]; 
 	
-	_artilleryunit = (allUnits select {_x distance _pos < 10 && _x != _civilian}); 
-	_artgroups pushBack (_artilleryunit select 0);
-	[_artilleryunit,_village] spawn CHAB_fnc_fire_artillery;
-	(_artilleryunit select 0) addEventHandler ["Killed", {
+	_createdVehicle setVariable ["disableInsurgencyCleanup", true];
```

Comment:
Not needed anymore. I disabled insurgency cleanup. You can remove this

Reply:
👍